### PR TITLE
Onboard containerregistry for autogeneration

### PR DESCRIPTION
**Pull request by anthony-c-martin**
```
> azure-schema-generator@1.0.0 generate-single C:\github\azure-resource-manager-schemas\generator
> ts-node cmd/generatesingle "containerregistry/resource-manager"

[C:\Users\antmarti\AppData\Local\Temp\schm_azspc] executing: git fsck
[C:\Users\antmarti\AppData\Local\Temp\schm_azspc] executing: git reset -q .
[C:\Users\antmarti\AppData\Local\Temp\schm_azspc] executing: git checkout -q -- .
[C:\Users\antmarti\AppData\Local\Temp\schm_azspc] executing: git clean -q -fd
[C:\Users\antmarti\AppData\Local\Temp\schm_azspc] executing: git gc
[C:\Users\antmarti\AppData\Local\Temp\schm_azspc] executing: git fetch
[C:\Users\antmarti\AppData\Local\Temp\schm_azspc] executing: git checkout -q origin/master
[C:\github\azure-resource-manager-schemas\generator] executing: autorest-beta.cmd --azureresourceschema --azureresourceschema.multi-scope=true --output-folder=C:\Users\antmarti\AppData\Local\Temp\cpbfgl5fpyq --tag=all-api-versions --api-version=2016-06-27-preview --title=none C:\Users\antmarti\AppData\Local\Temp\schm_azspc\specification\containerregistry\resource-manager\readme.md
AutoRest code generation utility [version: 3.0.6116; node: v10.15.3, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      'C:\github\azure-resource-manager-schemas\generator\node_modules\@autorest\core\dist' (3.0.6121)
   Loading local AutoRest extension '@autorest/azureresourceschema' (C:\github\azure-resource-manager-schemas\generator\node_modules\@autorest\azureresourceschema)
[10.35 s] Generation Complete
================================================================================================================================
Filename: C:\Users\antmarti\AppData\Local\Temp\cpbfgl5fpyq\2016-06-27-preview\Microsoft.ContainerRegistry.json
Provider Namespace: Microsoft.ContainerRegistry
API Version: 2016-06-27-preview
Resource Types (Resource Group Scope):
- registries
================================================================================================================================
[C:\github\azure-resource-manager-schemas\generator] executing: autorest-beta.cmd --azureresourceschema --azureresourceschema.multi-scope=true --output-folder=C:\Users\antmarti\AppData\Local\Temp\dn9v1wo5wzu --tag=all-api-versions --api-version=2017-06-01-preview --title=none C:\Users\antmarti\AppData\Local\Temp\schm_azspc\specification\containerregistry\resource-manager\readme.md
AutoRest code generation utility [version: 3.0.6116; node: v10.15.3, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      'C:\github\azure-resource-manager-schemas\generator\node_modules\@autorest\core\dist' (3.0.6121)
   Loading local AutoRest extension '@autorest/azureresourceschema' (C:\github\azure-resource-manager-schemas\generator\node_modules\@autorest\azureresourceschema)
[13.57 s] Generation Complete
================================================================================================================================
Filename: C:\Users\antmarti\AppData\Local\Temp\dn9v1wo5wzu\2017-06-01-preview\Microsoft.ContainerRegistry.json
Provider Namespace: Microsoft.ContainerRegistry
API Version: 2017-06-01-preview
Resource Types (Resource Group Scope):
- registries
- registries/replications
- registries/webhooks
================================================================================================================================
[C:\github\azure-resource-manager-schemas\generator] executing: autorest-beta.cmd --azureresourceschema --azureresourceschema.multi-scope=true --output-folder=C:\Users\antmarti\AppData\Local\Temp\bh071o1xexj --tag=all-api-versions --api-version=2018-02-01-preview --title=none C:\Users\antmarti\AppData\Local\Temp\schm_azspc\specification\containerregistry\resource-manager\readme.md
AutoRest code generation utility [version: 3.0.6116; node: v10.15.3, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      'C:\github\azure-resource-manager-schemas\generator\node_modules\@autorest\core\dist' (3.0.6121)
   Loading local AutoRest extension '@autorest/azureresourceschema' (C:\github\azure-resource-manager-schemas\generator\node_modules\@autorest\azureresourceschema)
[13.27 s] Generation Complete
================================================================================================================================
Filename: C:\Users\antmarti\AppData\Local\Temp\bh071o1xexj\2018-02-01-preview\Microsoft.ContainerRegistry.json
Provider Namespace: Microsoft.ContainerRegistry
API Version: 2018-02-01-preview
Resource Types (Resource Group Scope):
- registries/buildTasks
- registries/buildTasks/steps
================================================================================================================================
[C:\github\azure-resource-manager-schemas\generator] executing: autorest-beta.cmd --azureresourceschema --azureresourceschema.multi-scope=true --output-folder=C:\Users\antmarti\AppData\Local\Temp\3y1ac0bl8ck --tag=all-api-versions --api-version=2019-05-01-preview --title=none C:\Users\antmarti\AppData\Local\Temp\schm_azspc\specification\containerregistry\resource-manager\readme.md
AutoRest code generation utility [version: 3.0.6116; node: v10.15.3, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      'C:\github\azure-resource-manager-schemas\generator\node_modules\@autorest\core\dist' (3.0.6121)
   Loading local AutoRest extension '@autorest/azureresourceschema' (C:\github\azure-resource-manager-schemas\generator\node_modules\@autorest\azureresourceschema)
[11.28 s] Generation Complete
================================================================================================================================
Filename: C:\Users\antmarti\AppData\Local\Temp\3y1ac0bl8ck\2019-05-01-preview\Microsoft.ContainerRegistry.json
Provider Namespace: Microsoft.ContainerRegistry
API Version: 2019-05-01-preview
Resource Types (Resource Group Scope):
- registries/scopeMaps
- registries/tokens
================================================================================================================================
[C:\github\azure-resource-manager-schemas\generator] executing: autorest-beta.cmd --azureresourceschema --azureresourceschema.multi-scope=true --output-folder=C:\Users\antmarti\AppData\Local\Temp\ln5a9iunzm --tag=all-api-versions --api-version=2019-06-01-preview --title=none C:\Users\antmarti\AppData\Local\Temp\schm_azspc\specification\containerregistry\resource-manager\readme.md
AutoRest code generation utility [version: 3.0.6116; node: v10.15.3, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      'C:\github\azure-resource-manager-schemas\generator\node_modules\@autorest\core\dist' (3.0.6121)
   Loading local AutoRest extension '@autorest/azureresourceschema' (C:\github\azure-resource-manager-schemas\generator\node_modules\@autorest\azureresourceschema)
[17.76 s] Generation Complete
================================================================================================================================
Filename: C:\Users\antmarti\AppData\Local\Temp\ln5a9iunzm\2019-06-01-preview\Microsoft.ContainerRegistry.json
Provider Namespace: Microsoft.ContainerRegistry
API Version: 2019-06-01-preview
Resource Types (Resource Group Scope):
- registries/tasks
================================================================================================================================
[C:\github\azure-resource-manager-schemas\generator] executing: autorest-beta.cmd --azureresourceschema --azureresourceschema.multi-scope=true --output-folder=C:\Users\antmarti\AppData\Local\Temp\mrigqbwkuoa --tag=all-api-versions --api-version=2017-03-01 --title=none C:\Users\antmarti\AppData\Local\Temp\schm_azspc\specification\containerregistry\resource-manager\readme.md
AutoRest code generation utility [version: 3.0.6116; node: v10.15.3, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      'C:\github\azure-resource-manager-schemas\generator\node_modules\@autorest\core\dist' (3.0.6121)
   Loading local AutoRest extension '@autorest/azureresourceschema' (C:\github\azure-resource-manager-schemas\generator\node_modules\@autorest\azureresourceschema)
[12.52 s] Generation Complete
================================================================================================================================
Filename: C:\Users\antmarti\AppData\Local\Temp\mrigqbwkuoa\2017-03-01\Microsoft.ContainerRegistry.json
Provider Namespace: Microsoft.ContainerRegistry
API Version: 2017-03-01
Resource Types (Resource Group Scope):
- registries
================================================================================================================================
[C:\github\azure-resource-manager-schemas\generator] executing: autorest-beta.cmd --azureresourceschema --azureresourceschema.multi-scope=true --output-folder=C:\Users\antmarti\AppData\Local\Temp\67s6c2galkc --tag=all-api-versions --api-version=2017-10-01 --title=none C:\Users\antmarti\AppData\Local\Temp\schm_azspc\specification\containerregistry\resource-manager\readme.md
AutoRest code generation utility [version: 3.0.6116; node: v10.15.3, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      'C:\github\azure-resource-manager-schemas\generator\node_modules\@autorest\core\dist' (3.0.6121)
   Loading local AutoRest extension '@autorest/azureresourceschema' (C:\github\azure-resource-manager-schemas\generator\node_modules\@autorest\azureresourceschema)
[14.88 s] Generation Complete
================================================================================================================================
Filename: C:\Users\antmarti\AppData\Local\Temp\67s6c2galkc\2017-10-01\Microsoft.ContainerRegistry.json
Provider Namespace: Microsoft.ContainerRegistry
API Version: 2017-10-01
Resource Types (Resource Group Scope):
- registries
- registries/replications
- registries/webhooks
================================================================================================================================
[C:\github\azure-resource-manager-schemas\generator] executing: autorest-beta.cmd --azureresourceschema --azureresourceschema.multi-scope=true --output-folder=C:\Users\antmarti\AppData\Local\Temp\xnnxlyjxw5r --tag=all-api-versions --api-version=2018-09-01 --title=none C:\Users\antmarti\AppData\Local\Temp\schm_azspc\specification\containerregistry\resource-manager\readme.md
AutoRest code generation utility [version: 3.0.6116; node: v10.15.3, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      'C:\github\azure-resource-manager-schemas\generator\node_modules\@autorest\core\dist' (3.0.6121)
   Loading local AutoRest extension '@autorest/azureresourceschema' (C:\github\azure-resource-manager-schemas\generator\node_modules\@autorest\azureresourceschema)
[16.52 s] Generation Complete
================================================================================================================================
Filename: C:\Users\antmarti\AppData\Local\Temp\xnnxlyjxw5r\2018-09-01\Microsoft.ContainerRegistry.json
Provider Namespace: Microsoft.ContainerRegistry
API Version: 2018-09-01
Resource Types (Resource Group Scope):
- registries/tasks
================================================================================================================================
[C:\github\azure-resource-manager-schemas\generator] executing: autorest-beta.cmd --azureresourceschema --azureresourceschema.multi-scope=true --output-folder=C:\Users\antmarti\AppData\Local\Temp\ep1w6ceg9b --tag=all-api-versions --api-version=2019-04-01 --title=none C:\Users\antmarti\AppData\Local\Temp\schm_azspc\specification\containerregistry\resource-manager\readme.md
AutoRest code generation utility [version: 3.0.6116; node: v10.15.3, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      'C:\github\azure-resource-manager-schemas\generator\node_modules\@autorest\core\dist' (3.0.6121)
   Loading local AutoRest extension '@autorest/azureresourceschema' (C:\github\azure-resource-manager-schemas\generator\node_modules\@autorest\azureresourceschema)
[17.53 s] Generation Complete
================================================================================================================================
Filename: C:\Users\antmarti\AppData\Local\Temp\ep1w6ceg9b\2019-04-01\Microsoft.ContainerRegistry.json
Provider Namespace: Microsoft.ContainerRegistry
API Version: 2019-04-01
Resource Types (Resource Group Scope):
- registries/tasks
================================================================================================================================
[C:\github\azure-resource-manager-schemas\generator] executing: autorest-beta.cmd --azureresourceschema --azureresourceschema.multi-scope=true --output-folder=C:\Users\antmarti\AppData\Local\Temp\3exy4wiw87y --tag=all-api-versions --api-version=2019-05-01 --title=none C:\Users\antmarti\AppData\Local\Temp\schm_azspc\specification\containerregistry\resource-manager\readme.md
AutoRest code generation utility [version: 3.0.6116; node: v10.15.3, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      'C:\github\azure-resource-manager-schemas\generator\node_modules\@autorest\core\dist' (3.0.6121)
   Loading local AutoRest extension '@autorest/azureresourceschema' (C:\github\azure-resource-manager-schemas\generator\node_modules\@autorest\azureresourceschema)
[16.4 s] Generation Complete
================================================================================================================================
Filename: C:\Users\antmarti\AppData\Local\Temp\3exy4wiw87y\2019-05-01\Microsoft.ContainerRegistry.json
Provider Namespace: Microsoft.ContainerRegistry
API Version: 2019-05-01
Resource Types (Resource Group Scope):
- registries
- registries/replications
- registries/webhooks
================================================================================================================================
```

### CODE REVIEW

**Comment by ifeoluwaokunoren** [comment]
Hello - @mnltejaswini , @ankurkhemani , @jaysterp 

To simplify the schema generation process, ARM is working on an auto-generation pipeline your team can onboard to.
This means once a swagger PR from the Azure REST-API spec repo is merged either for a new api-version or an update to an existing api-version – the auto generation pipeline would generate the required schemas for that api-version daily – so changes in swagger would be automatically picked up.
The required action is for you to kindly review this PR to ensure that all expected resources have been detected and generated as expected.

You can reach out to me & Anthony over Teams/Email if you have any questions.

**Comment by jaysterp** [comment]
@ifeoluwaokunoren LGTM for ACR Task API versions (`2018-09-01`, `2019-04-01`, and `2019-06-01`). I will leave @ankurkhemani to review the ACR versions and give the final sign off.

File: schemas/2019-05-01/Microsoft.ContainerRegistry.json [hunk 1]
```diff
@@ -0,0 +1,768 @@
...
+        }
+      },
+      "description": "The properties of a container registry."
+    },
+    "ReplicationProperties": {
+      "type": "object",
+      "properties": {},
+      "description": "The properties of a replication."
+    },
+    "RetentionPolicy": {
```
Comment (ankurkhemani) [inline schemas/2019-05-01/Microsoft.ContainerRegistry.json]: Missing property `lastUpdatedTime` 
[Here ](https://github.com/Azure/azure-rest-api-specs/blob/daa4465509ed3d43260a7d1bfe7e2ca0b70ac9b9/specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-05-01/containerregistry.json#L1627)is the property is current swagger:

Comment (anthony-c-martin) [reply schemas/2019-05-01/Microsoft.ContainerRegistry.json]: Note that `readOnly` properties are removed intentionally - they cannot be directly set with a resource PUT during template deployment.

**Pull request merged**